### ISSUE TITLE
[fbpick-fine] Add robust-center jitter augmentation with in-window supervision

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#54]
# [fbpick-fine] Add robust-center jitter augmentation with in-window supervision

## Background

`fbpick-fine` currently trains on local windows centered around a physics-derived center, usually `robust_pick_i`.

Recent site54 OOF evaluation showed:

```text
- robust_pick_i is the best current fine-window center overall
- trend_center_i / fine_center_i is not reliable enough as a replacement
- confidence-based switching did not improve consistently
- horonobe2020 is a known hard case, but current priority is a generic model that works for most regions
```

For now, fine should continue to use:

```yaml
window_center:
  npz_key: robust_pick_i
  fallback_npz_key: null
```

However, during inference, the robust center can still be offset from the true first break. The fine model should therefore see center perturbations during training.

This issue adds **training-only robust-center jitter augmentation** while keeping the current fine target/loss design.  
Out-of-window samples are not used as negative/no-pick samples in this issue.

---

## Goal

Add configurable jitter to the fine training window center.

Base center:

```text
center_i = robust_npz["robust_pick_i"]
```

Training augmentation:

```text
center_i_train = center_i + sampled_jitter
```

Supervision rule:

```text
Only train on jittered windows where GT first-break remains inside the local window.
```

Validation and inference remain deterministic:

```text
center_i_valid = center_i
center_i_infer = center_i
```

This is Step 1 to stabilize the current fine model before introducing a larger P/Noise fine variant.

---

## Recommended site54 config

```yaml
window_center:
  npz_key: robust_pick_i
  fallback_npz_key: null

center_augment:
  enabled: true
  train_only: true
  p_no_jitter: 0.7
  uniform_jitter_samples:
    - {prob: 0.20, lo: -32, hi: 32}
    - {prob: 0.08, lo: -64, hi: 64}
    - {prob: 0.02, lo: -128, hi: 128}
  clip_to_record: true
  require_fb_inside: true
```

Default config must preserve existing behavior:

```yaml
center_augment:
  enabled: false
  train_only: true
  p_no_jitter: 1.0
  uniform_jitter_samples: []
  clip_to_record: true
  require_fb_inside: true
```

---

### 1. Add `center_augment` config for fine training

Target candidate:

```text
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/fine/config.py
```

Add dataclasses similar to:

```python
@dataclass(frozen=True)
class FineUniformJitterCfg:
    prob: float
    lo: int
    hi: int


@dataclass(frozen=True)
class FineCenterAugmentCfg:
    enabled: bool
    train_only: bool
    p_no_jitter: float
    uniform_jitter_samples: tuple[FineUniformJitterCfg, ...]
    clip_to_record: bool
    require_fb_inside: bool
```

Add it to `FineTrainConfig`.

Validation rules:

```text
enabled is bool
train_only is bool
0.0 <= p_no_jitter <= 1.0
each uniform jitter prob >= 0.0
sum(p_no_jitter + all probs) must be > 0.0
lo and hi are integer sample offsets
lo <= hi
clip_to_record is bool
require_fb_inside is bool
```

Probabilities may be normalized internally.  
If normalized, document this behavior clearly.

For this issue, `require_fb_inside=false` should either be rejected or treated as unsupported.

Recommended:

```text
center_augment.require_fb_inside must be true in this issue.
```

---

### 2. Add jitter sampling helper

Target candidate:

```text
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/fine/build_dataset.py
```

Add a helper similar to:

```python
def sample_center_jitter(
    *,
    size: int,
    cfg: FineCenterAugmentCfg,
    py_random: random.Random,
) -> np.ndarray:
    ...
```

or use the existing dataset RNG type if the fine dataset already uses `np.random.Generator`.

Requirements:

```text
- no global np.random state
- no global random state
- fixed seed gives reproducible jitter
- output dtype is integer
- output shape matches the requested center subset shape
```

Mixture behavior:

```text
p_no_jitter:
  jitter = 0

uniform_jitter_samples:
  choose one component by probability
  sample integer jitter uniformly from [lo, hi], inclusive
```

---

### 3. Apply jitter only in fine training dataset

Target candidate:

```text
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/fine/build_dataset.py
```

The fine training dataset currently loads center values from robust npz using `window_center.npz_key`.

Apply jitter after selecting `center_subset` and before local window extraction.

Pseudo logic:

```python
center_subset = centers[indices]

if center_augment.enabled:
    jitter = sample_center_jitter(
        size=center_subset.shape[0],
        cfg=center_augment,
        py_random=...
    )
    center_subset = center_subset + jitter

    if center_augment.clip_to_record:
        center_subset = np.clip(center_subset, 0, n_samples_orig - 1)
```

Do not apply jitter in:

```text
- validation dataset
- labeled infer dataset
- raw fine inference
```

---

### 4. Keep current target/loss behavior

This issue does not add no-pick/noise supervision.

The target remains:

```text
fb_idx_view = fb_i - window_start_i
```

Only jittered windows where GT is inside the local window are used for supervised training.

If jitter causes GT to fall outside the local window:

```text
reject that sample
```

Do not include it as a negative sample in this issue.

---

### 5. Add rejection accounting

If a jittered sample is rejected because GT falls outside the window, track it.

Suggested counter:

```text
center_jitter
```

or, if implementation is simpler:

```text
local_window
```

Preferred max-trials error:

```text
failed to draw a valid sample within max_trials=2048;
rejections: empty=..., min_pick=..., fblc=..., local_window=..., center_jitter=...
```

This is important because aggressive jitter can silently reduce useful samples.

---

### 6. Update fine train builder wiring

Where `build_train_dataset` creates the fine training dataset, pass:

```text
center_augment config
```

into the training dataset.

Validation dataset should receive either:

```text
center_augment disabled
```

or no augmentation config at all.

---

### 7. Update example config

Target candidates:

```text
examples/config_train_fbpick_fine.yaml
proc/fbpick/site54/configs/config_train_fbpick_fine.yaml
```

Add:

```yaml
window_center:
  npz_key: robust_pick_i
  fallback_npz_key: null

center_augment:
  enabled: true
  train_only: true
  p_no_jitter: 0.7
  uniform_jitter_samples:
    - {prob: 0.20, lo: -32, hi: 32}
    - {prob: 0.08, lo: -64, hi: 64}
    - {prob: 0.02, lo: -128, hi: 128}
  clip_to_record: true
  require_fb_inside: true
```

Do not switch to `fine_center_i`.

---

### 8. Update docs

Document:

```text
- Fine training uses robust_pick_i as center by default.
- Center jitter is train-only.
- Jitter simulates realistic coarse/physics center error.
- Jitter does not affect validation or inference.
- Out-of-window jittered samples are rejected, not used as negative samples.
- No-pick / noise supervision is out of scope for this issue.
- Monitor center_jitter / local_window rejection rate.
```

---

## Out of scope

This issue does not change:

```text
- fine model architecture
- fine output channels
- fine loss
- no-pick / noise class
- P/Noise fine variant
- fine inference behavior
- physics output schema
- robust_pick_i semantics
- trend_center_i or fine_center_i behavior
- fine window size
- horonobe-specific handling
```

No-pick/noise supervision should be handled in a separate issue, e.g.:

```text
[fbpick-fine] Add P/Noise local segmentation for no-pick fine windows
```

---

## Acceptance criteria

- Fine training config supports `center_augment`.
- Default config preserves existing behavior.
- Training dataset applies jitter to the configured window center when enabled.
- Validation dataset does not apply jitter.
- Fine inference does not apply jitter.
- Jitter supports multiple uniform components.
- Jitter sampling is reproducible with fixed seed.
- Jittered centers are clipped to record bounds when configured.
- Jittered samples where GT falls outside the local window are rejected.
- Out-of-window samples are not used as negative/no-pick samples.
- Rejection counters include jitter/local-window failures.
- Example fine training config is updated.
- Docs are updated.
- Unit tests are added.
- Existing tests pass.

---

### 1. Default config preserves behavior

Config:

```yaml
center_augment:
  enabled: false
```

Expected:

```text
center_subset == robust_pick_i subset
```

---

### 2. Constant jitter shifts training center

Robust centers:

```text
robust_pick_i = [100, 100, 100]
```

Config:

```yaml
center_augment:
  enabled: true
  p_no_jitter: 0.0
  uniform_jitter_samples:
    - {prob: 1.0, lo: 10, hi: 10}
  clip_to_record: true
  require_fb_inside: true
```

Expected:

```text
center_raw_i == 110
```

or local window start reflects center 110.

---

### 3. Validation does not apply jitter

Use the same config and robust file.

Expected validation behavior:

```text
center_raw_i == 100
```

---

### 4. Fine inference does not apply jitter

Fine inference should ignore `center_augment`.

Expected:

```text
window center is robust_pick_i without jitter
```

---

### 5. Center clipping

Input:

```text
center_i = 5
jitter = -100
n_samples_orig = 1000
clip_to_record = true
```

Expected:

```text
center_i_train == 0
```

---

### 6. Out-of-window jitter rejection

Fine window:

```text
time_len = 256
center_index = 128
```

Input:

```text
fb_i = 500
center_i = 500
jitter = 300
```

Expected:

```text
GT is outside local window
sample is rejected when require_fb_inside=true
center_jitter or local_window rejection counter increments
```

---

### 7. Reproducibility

With a fixed seed and same dataset access pattern:

```text
sampled jitter sequence is reproducible
```

With a different seed:

```text
jitter can differ
```

---

### 8. Probability mixture

Config:

```yaml
p_no_jitter: 0.5
uniform_jitter_samples:
  - {prob: 0.5, lo: 10, hi: 10}
```

Over many samples, both zero jitter and +10 jitter should occur.

Use a deterministic mocked RNG if approximate probabilistic tests are undesirable.

---

## Notes

This is intentionally conservative.

The current goal is to stabilize the existing fine model with robust-center jitter while keeping the supervised target/loss unchanged.

A future P/Noise fine task can use out-of-window jittered samples as negative/no-pick examples, but that requires a larger model/loss change and is out of scope here.
